### DISCUSSION TITLE
Running and rebuilding for dev purposes

### DISCUSSION THREAD

**Discussion by fguarini** [apache/superset#12925] [Q&A / Help] [answered]
Has been some time since I dived into supersets build process, I remember that in the past docker-compose used to trigger the build of the image, which would take into account any changes made.

Now that docker-compose reads directly from the dev image (apache/superset:latest-dev), does one have to just use the provided Dockerfile and tag it with the specific "apache/superset:latest-dev" or is there another simpler way that I am missing and that is intended to simplify the build process for dev purposes?

Thanks!

**Comment by srinify** [comment, marked as answer]
Hey @fguarini I'd actually recommend avoiding Docker Compose for development purposes. Docker is somewhat slow and makes your iteration cycles longer / more painful. In addition, Docker also tries to cache in many ways that can hinder development speed.

This is the workflow I generally recommend: https://github.com/apache/superset/blob/master/CONTRIBUTING.md#setup-local-environment-for-development

**Reply by fguarini** [reply]
Thanks a lot for the reply, sorry that I missed the documentation as you explicitly had instructions on it. Again thanks for pointing me to the right direction.
